frontend/src/app/pages/admin-page/admin-page.component.ts

```diff
@@ -1,10 +1,99 @@
-import { Component } from '@angular/core';
+import { Component, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import { ItemBackendService } from "../../services/item-backend.service";
+import { StatsBackendService } from "../../services/stats-backend.service";
+import { Item } from "../../models/item";
+import { Stats } from "../../models/stats";
+import { Subject, takeUntil } from "rxjs";
+import { MatTable, MatTableDataSource } from "@angular/material/table";
+import { MatPaginator, PageEvent } from "@angular/material/paginator";
+import { SnackBarService } from "../../services/snack-bar.service";
 
 @Component({
   selector: 'app-admin-page',
   templateUrl: './admin-page.component.html',
   styleUrls: ['./admin-page.component.css']
 })
-export class AdminPageComponent {
+export class AdminPageComponent implements OnInit, OnDestroy {
 
+  lastFiveItemsList: Item[];
+
+  dataSource = new MatTableDataSource<Stats>();
+  displayedColumns: string[] = ['category', 'itemsCount'];
+
+  totalElements: number;
+
+  currentPage: number = 0;
+  currentSize: number = 5;
+
+  loading: boolean;
+
+  @ViewChild(MatTable) table: any;
+  @ViewChild(MatPaginator) matPaginator: MatPaginator;
+
+  private unsubscribe: Subject<void> = new Subject();
+
+  constructor(private itemService: ItemBackendService,
+              private statsService: StatsBackendService,
+              private snackBarService: SnackBarService) {
+  }
+
+  ngOnInit() {
```

Comment:
you have in `ngOnInit()` to parrallel calls to backend, but you use only one "loading" boolean field. this leads to broken  display logic of separate elements of the page